File: client/nodejs/src/pdu/impl/AuthenticatedMessage.js
/**
 * Copyright 2016 TIBCO Software Inc. All rights reserved.
 * 
 * Licensed under the Apache License, Version 2.0 (the "License"); You may not use this file except 
 * in compliance with the License.
 * A copy of the License is included in the distribution package with this file.
 * You also may obtain a copy of the License at http://www.apache.org/licenses/LICENSE-2.0
 * 
 * Unless required by applicable law or agreed to in writing, software
 * distributed under the License is distributed on an "AS IS" BASIS,
 * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
 * See the License for the specific language governing permissions and
 * limitations under the License.
 */

var util                    = require('util'),
    AbstractProtocolMessage = require('../AbstractProtocolMessage').AbstractProtocolMessage,
    TGException             = require('../../exception/TGException').TGException,
    TGLogManager            = require('../../log/TGLogManager'),
    TGLogLevel              = require('../../log/TGLogger').TGLogLevel;

var logger = TGLogManager.getLogger();

function AuthenticatedMessage() {
	AuthenticatedMessage.super_.call(this);
    this._connectionId = null;
    this._clientId = null;
}

util.inherits(AuthenticatedMessage, AbstractProtocolMessage);

AuthenticatedMessage.prototype.getConnectionId = function() {
    return this._connectionId;
};

AuthenticatedMessage.prototype.setConnectionId = function(connectionId) {
    this._connectionId = connectionId;
};

AuthenticatedMessage.prototype.getClientId = function () {
    return this._clientId;
};

AuthenticatedMessage.prototype.setClientId = function (clientId) {
    this._clientId = clientId;
};

AuthenticatedMessage.prototype.writePayload = function (outputStream) {
	logger.logDebugWire( 
			"**** Entering commit AuthenticatedMessage.writePayload at output buffer position at : %d", 
			outputStream.getPosition());
    if ((this._authToken === -1) || (this._sessionId === -1)) {
        throw new Error("Message not authenticated");
    }

    outputStream.writeLong(this._authToken);
    outputStream.writeLong(this._sessionId);
    logger.logDebugWire( 
    		"**** Leaving commit AuthenticatedMessage.writePayload at output buffer position at : %d", 
    		outputStream.getPosition());
};

AuthenticatedMessage.prototype.readPayload = function (inputStream) {
    this.authToken = inputStream.readLong();
    this.sessionId = inputStream.readLong();
};

exports.AuthenticatedMessage = AuthenticatedMessage;

File: client/java/src/com/tibco/tgdb/pdu/impl/AuthenticatedMessage.java
/**
 * Copyright 2016 TIBCO Software Inc. All rights reserved.
 * 
 * Licensed under the Apache License, Version 2.0 (the "License"); You may not use this file except 
 * in compliance with the License.
 * A copy of the License is included in the distribution package with this file.
 * You also may obtain a copy of the License at http://www.apache.org/licenses/LICENSE-2.0
 * 
 * Unless required by applicable law or agreed to in writing, software
 * distributed under the License is distributed on an "AS IS" BASIS,
 * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
 * See the License for the specific language governing permissions and
 * limitations under the License.
 * <p/>
 * File name : AuthenticatedMessage.${EXT}
 * Created on: 2/4/15
 * Created by: suresh 
 * <p/>
 * SVN Id: $Id: AuthenticatedMessage.java 583 2016-03-15 02:02:39Z vchung $
 */


package com.tibco.tgdb.pdu.impl;

import com.tibco.tgdb.exception.TGException;
import com.tibco.tgdb.pdu.TGAuthenticatedMessage;
import com.tibco.tgdb.pdu.TGInputStream;
import com.tibco.tgdb.pdu.TGOutputStream;

import java.io.IOException;

public abstract class AuthenticatedMessage extends AbstractProtocolMessage implements TGAuthenticatedMessage {

//    long authToken;
//    long sessionId;
    int connectionId;
    String clientId;

    /*
    public long getAuthToken() {
        return authToken;
    }
    */

    /*
    public int getConnectionId() {
        return connectionId;
    }

    public void setConnectionId(int connectionId) {
        this.connectionId = connectionId;
    }

    public void setAuthToken(long authToken) {
        this.authToken = authToken;
    }

    public long getSessionId() {
        return sessionId;
    }

    public void setSessionId(long sessionId) {
        this.sessionId = sessionId;
    }
    */

    public String getClientId() {
        return clientId;
    }

    public void setClientId(String clientId) {
        this.clientId = clientId;
    }

    //FIXME: Is it ok to expose this?
    //Expose this so that it can be set
    //by the channel during sendRequest
    public void setRequestId(long requestId) {
    	this.requestId = requestId;
    }

    @Override
    protected void writePayload(TGOutputStream os) throws TGException, IOException {
        if ((authToken == -1) || (sessionId == -1))
            throw new TGException("Message not authenticated");

        os.writeLong(authToken);
        os.writeLong(sessionId);

    }

    @Override
    protected void readPayload(TGInputStream is) throws TGException, IOException {
        this.authToken = is.readLong();
        this.sessionId = is.readLong();
    }
}
